stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/service/sync/PushMessageSyncHandler.kt

```diff
@@ -73,31 +79,31 @@ public class PushMessageSyncHandler(private val service: Service) {
             return
         }
         createSyncNotificationChannel()
-        showForegroundNotification(notificationConfig.smallIcon)
+        startForegroundExecution(notificationConfig.smallIcon)
 
-        GlobalScope.launch(DispatcherProvider.IO) {
```

Comment:
Since it's a service there's no need to run sync on IO dispatcher. Running sync operation synchronously will prevent potential side effects (manual switching background/foreground mode by the user during sync operation).

Reply:
Services run on the main thread by default ([see here](https://developer.android.com/guide/components/services)), but the `onMessageReceived` method in the Firebase SDK does promise to be on a worker thread.

Reply:
Might be worth adding `@WorkerThread` annotations to these methods, if we expect them to always be called from a background thread

Reply:
I've added annotation to `onMessageReceived` and modified the sync call to run synchronously within the thread it's called.